### PR TITLE
NFDIV-1252: Refine ES queries so that no redundant results are returned

### PR DESCRIPTION
**Pull request by tomxelliott**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-1252

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/systemupdate/schedule/SystemProgressCasesToAosOverdueTask.java [hunk 1]
```diff
@@ -49,7 +49,8 @@ public void run() {
         final String serviceAuth = authTokenGenerator.generate();
 
         try {
-            final List<CaseDetails> casesInAwaitingAosState = ccdSearchService.searchForAllCasesWithStateOf(AwaitingAos, user, serviceAuth);
+            final List<CaseDetails> casesInAwaitingAosState =
+                ccdSearchService.searchForAllCasesWithStateOf(AwaitingAos, null, user, serviceAuth);
```
Comment (mapingo) [inline src/main/java/uk/gov/hmcts/divorce/systemupdate/schedule/SystemProgressCasesToAosOverdueTask.java:53]: Maybe there should be two search methods, one that has the flag parameter and one without.  That way we don't have to pass the null as an argument, internally the methods can call a single implementation.

File: src/main/java/uk/gov/hmcts/divorce/systemupdate/service/CcdSearchService.java [hunk 2]
```diff
@@ -62,15 +65,44 @@ public SearchResult searchForCaseWithStateOf(final State state,
             sourceBuilder.toString());
     }
 
-    public List<CaseDetails> searchForAllCasesWithStateOf(final State state, User user, String serviceAuth) {
+    public SearchResult searchForCasesWithStateOfDueDateBeforeWithoutFlagSet(final State state,
+                                                                             final int from,
+                                                                             final int size,
+                                                                             final String flag,
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/systemupdate/service/CcdSearchService.java:71]: would be nice to call this as searchQueryParam or something similar

Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/divorce/systemupdate/service/CcdSearchService.java:71]: Sorry what do you mean?

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/systemupdate/service/CcdSearchService.java]: I mean instead of flag we could call it searchQueryParam :)

**Review by nitinprabhu — APPROVED** [review]

**Pull request merged**